python/ray/external_storage.py

```diff
@@ -76,32 +200,33 @@ class FileSystemStorage(ExternalStorage):
 
     def __init__(self, directory_path):
         self.directory_path = directory_path
-        self.prefix = "ray_spilled_object_"
+        self.prefix = DEFAULT_OBJECT_PREFIX
         os.makedirs(self.directory_path, exist_ok=True)
         if not os.path.exists(self.directory_path):
             raise ValueError("The given directory path to store objects, "
                              f"{self.directory_path}, could not be created.")
-
-    def spill_objects(self, object_refs):
-        keys = []
-        ray_object_pairs = self._get_objects_from_store(object_refs)
-        for ref, (buf, metadata) in zip(object_refs, ray_object_pairs):
-            filename = self.prefix + ref.hex()
-            with open(os.path.join(self.directory_path, filename), "wb") as f:
-                metadata_len = len(metadata)
-                buf_len = len(buf)
-                f.write(metadata_len.to_bytes(8, byteorder="little"))
-                f.write(buf_len.to_bytes(8, byteorder="little"))
-                f.write(metadata)
-                f.write(memoryview(buf))
-            keys.append(filename.encode())
-        return keys
+        self.url_protocol = ExternalStorageURLProtocol(self.prefix)
+
+    def spill_objects(self, object_refs) -> List[str]:
+        if len(object_refs) == 0:
+            return []
+        # Always use the first object ref as a key when fusioning objects.
+        first_ref = object_refs[0]
+        object_id_stored = first_ref.hex()
+        filename = self.url_protocol.produce_external_url_from_object_ref(
```

Comment:
Why not keep `filename = self.prefix + ref.hex()`? That's easier to understand.

Reply:
Could also do
```
filename = self.prefix + ref.hex() + "-multi-{}".format(len(object_refs))
```
to make it more clear there are multiple objects stored.

Reply:
The second one looks good. I will go with that approach.